<h2>Quiz 1 : Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [0]:
# Membuat dataframes dari dictionary
import pandas as pd

obj_dict = {
    'keluarga' : ['Ayah', 'Ibu', 'Kakak'],
    'suhu' : [36.6, 36.7, 37.0]    
}

df = pd.DataFrame(obj_dict)
print(df)

  keluarga  suhu
0     Ayah  36.6
1      Ibu  36.7
2    Kakak  37.0


<h2>Quiz 2 : Membaca data dari file</h2>

Download file bernama diamonds.csv, kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [0]:
# Membaca data dan memasukannya kedalam suatu variabel
df_csv = pd.read_csv('diamonds.csv')
df_csv

Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]

Expected Result :

![image.png](attachment:image.png)

<h2>Quiz 3 : DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Perbedaan Series dengan list yaitu terletak pada indexnya, dimana jika list hanya bisa int sedangkan index pada series dapat diubah semisal menjadi string. Adapun dengan dictionary yaitu terletak pada duplikasi keynya yang dictionary tidak bisa duplikasi adanya key yang sama sedangkan series dapat melakukan hal tersebut.

Perbedaan dataframe dengan series terletak pada dimensinya, jika series 1 dimensi sedangkan dataframe 2 dimensi

In [0]:
# code here
obj_series = pd.Series([99, 98, 97], index=['x', 'y', 'z'])
obj_series

x    99
y    98
z    97
dtype: int64

<h2>Quiz 4 : Pemeriksaan Data Sederhana</h2>

Di quiz no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari :
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [0]:
# code here
df_csv.head(1)


Unnamed: 0  carat    cut color clarity  ...  table  price     x     y     z
0           1   0.23  Ideal     E     SI2  ...   55.0    326  3.95  3.98  2.43

[1 rows x 11 columns]

In [0]:
df_csv.tail(1)

Unnamed: 0  carat    cut color clarity  ...  table  price     x     y     z
53939       53940   0.75  Ideal     D     SI2  ...   55.0   2757  5.83  5.87  3.64

[1 rows x 11 columns]

baris pertama dari column price : 326

baris terakhir dari column color : D

<h2>Quiz 5 : Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Statistika deskriptif merupakan suatu metode yang berhubungan dengan pengumpulan atau penyajian data sampai memberi informasi yang berguna.

<h2>Quiz 6 : Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [0]:
# code here
df_csv.describe()

Unnamed: 0         carat  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean   26970.500000      0.797940  ...      5.734526      3.538734
std    15571.281097      0.474011  ...      1.142135      0.705699
min        1.000000      0.200000  ...      0.000000      0.000000
25%    13485.750000      0.400000  ...      4.720000      2.910000
50%    26970.500000      0.700000  ...      5.710000      3.530000
75%    40455.250000      1.040000  ...      6.540000      4.040000
max    53940.000000      5.010000  ...     58.900000     31.800000

[8 rows x 8 columns]

Berapa nilai rata-rata dari column price : 3932.799722

Berapa nilai standar deviasi dari column depth : 1.432621

Berapa nilai maximum dari column carat : 5.010000

# New Section